pkg/ebpf/uprobes/attacher.go

```diff
@@ -0,0 +1,721 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package uprobes
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	manager "github.com/DataDog/ebpf-manager"
+	"golang.org/x/exp/maps"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/sharedlibraries"
+	"github.com/DataDog/datadog-agent/pkg/network/usm/utils"
+	"github.com/DataDog/datadog-agent/pkg/process/monitor"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+// ExcludeMode defines the different optiont to exclude processes from attachment
+type ExcludeMode uint8
+
+const (
+	// ExcludeSelf excludes the agent's own PID
+	ExcludeSelf ExcludeMode = 1 << iota
+	// ExcludeInternal excludes internal DataDog processes
+	ExcludeInternal
+	// ExcludeBuildkit excludes buildkitd processes
+	ExcludeBuildkit
+	// ExcludeContainerdTmp excludes containerd tmp mounts
+	ExcludeContainerdTmp
```

Comment:
That's USM specific, I'm not sure it should be defined in a generic package

Reply:
From what I understood, it seemed like excluding docker build processes might be useful for all probes, no?

Reply:
it depends on the probe itself, so I'm not sure if it should be in the infrastructure

Reply:
I thought about having a generic interface to exclude certain binaries, but the problem then is when to run those filters: we should run them as soon as possible to avoid extra load, and that point depends on what the filter looks at. I felt that just for these two checks that maybe could be reused it's ok, but I can also make those filters and check them at the beggining of `attachToBinary` as an initial approach, to be modified if needed later.